File: src/node/magma/None.ts






class None<T> {
	;
	constructor () {
	}
	ifPresent(consumer : Consumer<T>) : void {
	}
	isPresent() : boolean {
		return false;
	}
	or(other : Supplier<Optional<T>>) : Optional<T> {
		return other.get();
	}
	orElseGet(other : Supplier<T>) : T {
		return other.get();
	}
	map<R>(mapper : Function<T, R>) : Optional<R> {
		return new None();
	}
	flatMap<R>(mapper : Function<T, Optional<R>>) : Optional<R> {
		return new None();
	}
	orElse(other : T) : T {
		return other;
	}
	filter(predicate : Predicate<T>) : Optional<T> {
		return this;
	}
	isEmpty() : boolean {
		return true;
	}
	toTuple(other : T) : Tuple<Boolean, T> {
		return new Tuple(false, other);
	}
	stream() : Stream<T> {
		return Stream.empty();
	}
}

File: src/java/magma/None.java
package magma;

import java.util.function.Consumer;
import java.util.function.Function;
import java.util.function.Supplier;
import java.util.stream.Stream;

class None<T> implements Option<T> {

	@Override
	public final <R> Option<R> map(final Function<T, R> mapper) {
		return new None<>();
	}

	@Override
	public final T orElse(final T other) {
		return other;
	}

	@Override
	public void ifPresent(final Consumer<T> consumer) {
	}

	@Override
	public final boolean isEmpty() {
		return true;
	}

	@Override
	public final Option<T> or(final Supplier<Option<T>> other) {
		return other.get();
	}

	@Override
	public final T orElseGet(final Supplier<T> other) {
		return other.get();
	}

	@Override
	public final <R> Option<R> flatMap(final Function<T, Option<R>> mapper) {
		return new None<>();
	}

	@Override
	public final Stream<T> stream() {
		return Stream.empty();
	}

	@Override
	public final Tuple<Boolean, T> toTuple(final T other) {
		return new Tuple<>(false, other);
	}
}
